### PR TITLE
add_piece_position_funcs

### PR DESCRIPTION
**Pull request by vkantsev**

### CODE REVIEW

File: src/state.py [hunk 1]
```diff
@@ -39,82 +39,95 @@ class Board:
...
     # board shape and constants with their defaults for a square board
     # with side 14 positions
-    shape_angles: int = 4  # rectangular = 4, triangular = 3, pentagon = 5
-    shape_side_length: int = 14  # side length of the board shape
+    board_corners: int = 4  # rectangular = 4, triangular = 3, pentagon = 5
+    board_side_length: int = 14  # side length of the board shape
 
     # the offset between two neighbor players start possitions on the board
-    player_shift: int = 0
+    player_shift: int = (board_side_length + 1)
```
[the remark this answers is not in the document]
Comment (vkantsev) [reply src/state.py:51]: `player_shift` can replace `create()`?

File: src/state.py [hunk 2]
```diff
@@ -39,82 +39,95 @@ class Board:
...
+            relative_position
+            if relative_position <= self.path_zone_length
+            else relative_position % self.path_zone_length
+        )
+        return relative_position
 
-    # TODO finish off implementation and add unit tests
     def is_on_start(self, piece: Piece) -> bool:
-        return piece.position == 0
+        return True if piece.position == 1 else False
```
[the remark this answers is not in the document]
Comment (vkantsev) [reply src/state.py:115]: Good point. I considered _start_ to be the first position on the _path_ vs _home_ for `0`.

File: src/state.py [hunk 3]
```diff
@@ -39,82 +39,95 @@ class Board:
...
 
     # TODO finish off implementation and add unit tests
     def is_on_path(self, piece: Piece) -> bool:
-        relative_position = self.relative_position(piece)
-        end_path = self.end_progress - self.finish_zone_length
-        return piece.position > 0 and relative_position < end_path
+        return True if 1 <= piece.position <= self.path_zone_length else False
```
[the remark this answers is not in the document]
Comment (vkantsev) [reply src/state.py:118]: * Nicely shortened;
* Fixed logic;

File: src/state_test.py [hunk 4]
```diff
@@ -64,18 +65,19 @@ def test_2_players_board_init(monkeypatch):
...
     assert board.pieces_per_player == 4
-    assert board.shape_angles == 5
-    assert board.shape_side_length == 14
+    assert board.board_corners == 5
+    assert board.board_side_length == 14
     assert board.player_shift == 15
+    assert board.path_zone_length == 5 * 14 + 5
     assert board.finish_zone_length == 5
-    assert board.end_progress == 5 * 14 + 1 + 5
+    assert board.end_progress == 5 * 14 + 5 + 5 + 1
```
[the remark this answers is not in the document]
Comment (vkantsev) [reply src/state_test.py:80]: Added comments for clarity

File: src/state_test.py [hunk 5]
```diff
@@ -158,3 +161,41 @@ def test_game_state_defaults(monkeypatch):
...
+
+    rel_pos_p3 = board.relative_position(piece=Piece(number=0, player=3, position=20))
+    assert rel_pos_p3 == 5
+
+    # Test a position outside of path_zone
+    with pytest.raises(Exception):
+        board.relative_position(piece=Piece(number=0, player=0, position=61))
+
+
+def test_board_is_on_start():
```
[the remark this answers is not in the document]
Comment (vkantsev) [reply src/state_test.py:194]: Added `is_on_path` , `is_on_finish`, `is_on_target` tests. Does `is_on_end` == `is_on_target`?

File: src/state.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (luchesar) [reply src/state.py:50]: Not exactly but close. The ability to initialise variables based on other variables passed in the constructor gives us the option to remove the `create()` where this initialisation is currently done.

File: src/state_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (luchesar) [reply src/state_test.py:194]: Ah yes, that's what I meant great!

**Pull request merged**